File: BotSearcher.java
// 🛡️ 0mega Code: Hybrid Bot-Scanner v1.0
// Developed by 0mega-One
// File: BotSearcher.java

import java.util.ArrayList;

public class BotSearcher { // El nombre de la clase ahora coincide con el archivo
    private String username;
    private int suspicionScore = 0;

    public BotSearcher(String username) {
        this.username = username;
    }

    public void analyzeChat(String message) {
        String[] blacklist = {".com", ".gg", "robux", "v e n t a", "c0m"};
        String lowerMessage = message.toLowerCase();

        for (String word : blacklist) {
            if (lowerMessage.contains(word)) {
                this.suspicionScore += 2;
            }
        }
    }

    public void analyzeMovement(double distance, boolean isAntiAfk) {
        if (distance < 0.01 || isAntiAfk) {
            this.suspicionScore += 1;
        }
    }

    public String getVerdict() {
        if (this.suspicionScore >= 3) {
            return "🚨 PROTOCOL 0MEGA-ONE: Bot confirmed. KICK " + this.username;
        }
        return "✅ User " + this.username + " is safe.";
    }

    public static void main(String[] args) {
        BotSearcher scanner = new BotSearcher("BotTester_Java");
        scanner.analyzeChat("Free items at scam.c0m");
        scanner.analyzeMovement(0.0, true);
        System.out.println(scanner.getVerdict());
    }
}

File: BotSearcher.cs
// 🛡️ 0mega Code: Hybrid Bot-Scanner v1.0
// Developed by 0mega-One
// Specialized for Unity and .NET Game Servers

using System;
using System.Collections.Generic;
using System.Linq;

namespace OmegaCode.Security
{
    public class BotScanner
    {
        public string Username { get; private set; }
        private int _suspicionScore = 0;

        public BotScanner(string username)
        {
            Username = username;
        }

        public void AnalyzeChat(string message)
        {
            // Lista de detección de 0mega-One
            string[] blacklist = { ".com", ".gg", "robux", "v e n t a", "c0m" };
            
            foreach (var word in blacklist)
            {
                if (message.ToLower().Contains(word))
                {
                    _suspicionScore += 2;
                    Console.WriteLine($"[!] {Username}: Link or suspicious keyword detected.");
                }
            }
        }

        public void AnalyzeMovement(double velocity, bool isAntiAfkActive)
        {
            // Si el movimiento es demasiado constante o nulo durante el spam
            if (velocity < 0.01 || isAntiAfkActive)
            {
                _suspicionScore += 1;
                Console.WriteLine($"[!] {Username}: Robotic movement pattern detected.");
            }
        }

        public string GetFinalVerdict()
        {
            if (_suspicionScore >= 3)
            {
                return $"🚨 PROTOCOL 0MEGA-ONE: Bot confirmed. Kick user {Username}.";
            }
            return $"✅ User {Username} is safe for now.";
        }
    }

    // Ejemplo de ejecución
    class Program
    {
        static void Main()
        {
            BotScanner myScanner = new BotScanner("BloxVents_Unity");
            myScanner.AnalyzeChat("Get free R-Bux at Scam.c0m");
            myScanner.AnalyzeMovement(0.0, true);

            Console.WriteLine(myScanner.GetFinalVerdict());
        }
    }
}
